core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/OutputExpression.scala

```diff
@@ -0,0 +1,31 @@
+package org.broadinstitute.dsde.rawls.expressions
+
+import org.broadinstitute.dsde.rawls.{RawlsException, StringUtils}
+import org.broadinstitute.dsde.rawls.model.{AttributeName, ErrorReportSource}
+
+sealed trait OutputExpressionTarget {
+  val prefix: String
+  def remainder(s: String): String = s.stripPrefix(prefix)
+}
+case object ThisEntityTarget extends OutputExpressionTarget { override val prefix = "this." }
+case object WorkspaceTarget extends OutputExpressionTarget { override val prefix = "workspace." }
+
+case class OutputExpression(target: OutputExpressionTarget, attributeName: AttributeName)
+
+object OutputExpression extends StringUtils {
+  override implicit val errorReportSource = ErrorReportSource("rawls")
+
+  def apply(s: String): OutputExpression = {
+    val target = if(s.startsWith(ThisEntityTarget.prefix))
+      ThisEntityTarget
+    else if(s.startsWith(WorkspaceTarget.prefix))
+      WorkspaceTarget
+    else
+      throw new RawlsException(s"Invalid output expression: $s")
+
+    val attributeName = AttributeName.fromDelimitedName(target.remainder(s))
+    validateUserDefinedString(attributeName.name)
```

Comment:
This is the same validation done later when we try to save attributes back to the data model